paasta_tools/vitesscluster_tools.py

```diff
@@ -15,6 +17,594 @@
+    """
+    Build vtgate config
+    """
+    config = {
+        "name": cell,
+        "gateway": {
+            "extraFlags": {
+                "mysql_auth_server_impl": "vault",
+                "mysql_auth_vault_addr": f"https://vault-dre.{region}.yelpcorp.com:8200",
+                "mysql_auth_vault_path": "secrets/vitess/vt-gate/vttablet_credentials.json",
+                "mysql_auth_vault_tls_ca": f"/etc/vault/all_cas/acm-privateca-{region}.crt",
+                "mysql_auth_vault_ttl": "60s",
+            },
+            "affinity": build_affinity_spec(paasta_pool),
+            "extraLabels": build_extra_labels(paasta_pool, paasta_cluster),
+            "extraEnv": build_extra_env(paasta_cluster),
+            "replicas": 1,
+            "resources": {
+                "requests": {
+                    "cpu": "100m",
+                    "memory": "256Mi",
+                },
+                "limits": {"memory": "256Mi"},
+            },
+        },
+    }
+    vtgate_extra_env = copy.deepcopy(VTGATE_EXTRA_ENV)
+    for env in vtgate_extra_env:
+        if env["name"] == "VAULT_ADDR":
+            env["value"] = f"https://vault-dre.{region}.yelpcorp.com:8200"
+        if env["name"] == "VAULT_CACERT":
+            env["value"] = f"/etc/vault/all_cas/acm-privateca-{region}.crt"
+        config["gateway"]["extraEnv"].append(env)
+    return config
+
+
+def build_vitess_dashboard_config(cells, paasta_pool, paasta_cluster, zk_address):
+    """
+    Build vtctld config
+    """
+    config = {
+        "cells": cells,
+        "affinity": build_affinity_spec(paasta_pool),
+        "extraLabels": build_extra_labels(paasta_pool, paasta_cluster),
+        "extraEnv": build_extra_env(paasta_cluster),
+        "extraFlags": VTCTLD_EXTRA_FLAGS,
+        "replicas": 1,
+        "resources": {
+            "requests": {
+                "cpu": "100m",
+                "memory": "128Mi",
+            },
+            "limits": {"memory": "128Mi"},
+        },
+    }
+    vtctld_extra_env = copy.deepcopy(VTCTLD_EXTRA_ENV)
+    for env in vtctld_extra_env:
+        if env["name"] == "TOPOLOGY_FLAGS":
+            env[
+                "value"
+            ] = f"--topo_implementation {TOPO_IMPLEMENTATION} --topo_global_server_address ${zk_address} --topo_global_root {TOPO_GLOBAL_ROOT}"
+        config["extraEnv"].append(env)
+
+    return config
+
+
+def build_vt_admin_config(cells, paasta_pool, paasta_cluster):
+    """
+    Build vtadmin config
+    """
+    config = {
+        "cells": cells,
+        "apiAddresses": ["http://localhost:15000"],
+        "affinity": build_affinity_spec(paasta_pool),
+        "extraLabels": build_extra_labels(paasta_pool, paasta_cluster),
+        "extraFlags": VTADMIN_EXTRA_FLAGS,
+        "extraEnv": build_extra_env(paasta_cluster),
+        "replicas": 1,
+        "readOnly": False,
+        "apiResources": {
+            "requests": {
+                "cpu": "100m",
```

Comment:
Same here, have to modify resource specifications based on service metadata in yelpsoa

Reply:
Fixed in https://github.com/Yelp/paasta/commit/6f56e5f7da82df3de44e035247fe865daf409ff4